src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -203,6 +203,28 @@ _get_available_port (void)
   return port;
 }
 
+/**
+ * @brief Internal function to check connection.
+ */
+static bool
+_nns_edge_check_connection (nns_edge_conn_s * conn)
+{
+  GIOCondition condition;
+
+  if (!conn || !conn->socket || g_socket_is_closed (conn->socket))
+    return false;
+
+  condition = g_socket_condition_check (conn->socket,
+      G_IO_IN | G_IO_OUT | G_IO_PRI | G_IO_ERR | G_IO_HUP);
```

Comment:
FYI,
> It is meaningless to specify G_IO_ERR or G_IO_HUP in condition; these conditions will always be set in the output if they are true.

https://libsoup.org/gio/GSocket.html#g-socket-condition-check

Reply:
Agreed, but for future work - remove glib :)